### ISSUE TITLE
🐛 Bug Report: Triggering an event for multiple receivers fails when one of them is invalid

### ISSUE THREAD

**Issue by JonasDoe** [notifirehq/notifire#6572]
### 📜 Description
**Update**: I've posted an updated and more precise description of the error in a comment [below](https://github.com/novuhq/novu/issues/6572#issuecomment-2396755352).

**Old description**:
I trigger an event with this payload:
```json
{
    "actor": "myTestActor",
    "name": "generic-message",
    "payload": {
        "myTestData": "hello"
    },
    "to": [
        {
            "type": "Topic",
            "topicKey": "myTopic"
        },
        {
            "type": "Subscriber",
            "subscriberId": "mySubscriber"
        }
    ],
}
```
`mySubscriber` will get a notification, subscribers to `myTopic` won't. Removing the second `to` entry won't help - the topic won't receive the event as long as `to` has the form of a list.

### 👟 Reproduction steps

1. Create a topic named `myExistingTopic` and subscribe with a user to it.
2. Run
```sh
curl --location 'https://api.notification.app.myCompany.cloud/v1/events/trigger' \
--header 'Authorization: ApiKey *******' \
--header 'Content-Type: application/json' \
--data '{
    "actor": "myTestActor",
    "name": "generic-message",
    "payload": {
        "myTestData": "hello"
    },
     "to": [
        {
            "type": "Topic",
            "topicKey": "myTopic"
        },
        {
            "type": "Subscriber",
            "subscriberId": "mySubscriber"
        }
    ],
}'
```
The response is `201` with
```json
{
    "data": {
        "acknowledged": true,
        "status": "processed",
        "transactionId": "487f708e-4a37-4438-9ec8-87deb5510ee2"
    }
}
```
3. See that the subscribers for `myTopic` didn't receive any messge, while the subscriber did. Furthermore, in the Novu Center's Activity Feed you won't see anything about `myTopic`.

### 👍 Expected behavior

The users subscribed to `myTopic` should receive a message about the triggered event.

### 👎 Actual Behavior with Screenshots

The users subscribed to `myTopic` doesn't receive a message about the triggered event.

### Novu version

0.24.0

### npm version

I cannot provide this information right now, but we're basically using ghcr.io/novuhq/novu/api:0.24.0

### node version

20.11.1

### 📃 Provide any additional context for the Bug.

It seems like triggering an event for a topic works when it's not in an array. So
```json
{
    ...
    "to": {
        "type": "Topic",
        "topicKey": "myTopic"
    }
}
```
will work.
### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find a similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

**Comment by JonasDoe** [comment]
I realized it's probalbly not cause by the topic's existence but by the mere fact I'm passing multiple topics - adjusted the issue accordingly.

**Comment by jainpawan21** [comment]
@JonasDoe 

This is correct format

```
"to": [
        {
            "type": "Topic",
            "topicKey": "first-topic"
        },
        {
            "subscriberId": "hello-first-subscriber"
        }
    ]
```

**Comment by JonasDoe** [comment]
> @JonasDoe
> 
> This is correct format
> 
> ```
> "to": [
>         {
>             "type": "Topic",
>             "topicKey": "first-topic"
>         },
>         {
>             "subscriberId": "hello-first-subscriber"
>         }
>     ]
> ```

Oh, I see, Then the error is more what I thought first - once an entry is invalid, nothing will be sent. I've tested again, even this won't work:
```json
...
"to": [
    {
        "type": "Topic",
        "topicKey": "first-topic"
    },
    {
        "type": "Topic",
        "topicKey": "non-existent-topic"
    },
]
```
So, subscribers of `first-topic` won't get anything if:
- there's another topic which does not exist
- there's a an unexpected `key: Subscriber`

In the first scenario I'ld expect to reveive a message from `first-topic`
In the second scenario I'ld expect either the same or get a `400`-response telling me about an invalid API usage.

**Comment by jainpawan21** [comment]
Yes you are right. if any entry is invalid. Notification is not delivered. This sounds like a bug.

**Comment by BadriVishalPadhy** [comment]
@JonasDoe can u assign this issue to me
